pkg/storage/mock/storer.go

```diff
@@ -6,28 +6,41 @@ package mock
 
 import (
 	"context"
+	"sync"
 
 	"github.com/ethersphere/bee/pkg/storage"
 	"github.com/ethersphere/bee/pkg/swarm"
 )
 
+type MockStorer interface {
```

Comment:
I am not sure if MockStorer interface is needed. Struct mockStore can be exported with all available methods that it provides. It just have to ensured that MockStore struct implements storage.Storer at compile time with `var _ storage.Storer = (*MockStorer)(nil)`. NewStorer would have to return a concrete type.